### PR TITLE
templates/openshift: grab bootstrap log on failure

### PR DESCRIPTION
**Pull request by crawford**
This uses the Terraform state to discover the IP address of the
bootstrap node (ideally, the installer will provide this information in
a form which easier to consume in the future). It then connects to the
gatewayd instance on that machine and pulls the logs for various
services. Hopefully, these logs will be useful when diagnosing
installation failures.

Continuing from https://github.com/openshift/release/pull/2624. Thanks @sttts!

This still requires a hole to be punched in the security group for gatewayd. I'm going to follow up next week.
/hold

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
@crawford this allows us to read container logs from bootstrap node kubelet as we opened `10250` [here](https://github.com/openshift/installer/pull/1105)
 
https://github.com/abhinavdahiya/release/commit/525c911e965f291120db52d7e05fd71694b5a0cb

also @smarterclayton do you think this is sane?

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml [hunk 1]
```diff
@@ -373,8 +373,7 @@ objects:
             exit 1
         fi
 
-        /bin/openshift-install --dir=/tmp/artifacts/installer create cluster &
-        wait "$!"
+        /bin/openshift-install --dir=/tmp/artifacts/installer create cluster
```
Comment (smarterclayton) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: Why this change?

**Comment by smarterclayton** [comment]
Not unreasonable, although obv in future it would be better to have integrated.  Maybe add a comment about that.

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 1 above)
Comment (crawford) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: That was an artifact of some previous refactoring. Backgrounding the process and then immediately waiting on it should be equivalent to just running it in the foreground.

**Comment by crawford** [comment]
Verified that this now works with the latest installer. Service logs get written into `/tmp/artifacts/bootstrap/`.

/hold cancel

File: ci-operator/templates/openshift/installer/cluster-launch-installer-src.yaml [hunk 2]
```diff
@@ -338,8 +338,7 @@ objects:
             exit 1
         fi
 
-        /bin/openshift-install --dir=/tmp/artifacts/installer create cluster &
-        wait "$!"
+        /bin/openshift-install --dir=/tmp/artifacts/installer create cluster
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-src.yaml]: Why did you touch this?  Won't it break the `/tmp/exit`-setting trap we set at the beginning of this script?  Previous discussion in d862f6f12958 (#1957).

Comment (wking) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-src.yaml]: ah, previous discussion [here][1].  I still think this change breaks the `trap`.

[1]: https://github.com/openshift/release/pull/2633/files#r249622809

File: ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml (hunk 1 above)
Comment (wking) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml]: I've filed #2680, rolling this back and adding an additional `wait` to our `TERM` handlers.  As discussed in that PR, we need the backgrounded `create cluster` so the shell will fire the `TERM` handler which `TERM`s the child processes.  As master stands (with this PR but without #2680), aborting an e2e job will `TERM` the setup container's shell process, but that process will not forward the `TERM` to `openshift-install` and will wait until it exits normally.  I haven't actually tested with a wrapping container, but my guess is that this means that jobs being aborted during `openshift-install` are mostly hitting their gentle-shutdown timeout and getting `KILL`ed.  With #2680, the forwarded `TERM` gives the installer a chance at a gentle shutdown (although at the moment, I don't think the installer takes advantage of that opportunity).
